### PR TITLE
US011 - Display state of a collection exercise

### PR DESCRIPTION
**Pull request by emilioward**
[https://trello.com/c/kJO9QkLd](https://trello.com/c/kJO9QkLd)

Added status column and data to the survey details page.

Data needs to be loaded in to test this functionality.

Don't merge until other user stories related to this card have been tested against this PR.
These states might change as they are, but the mapping functionality works.

### CODE REVIEW

File: response_operations_ui/templates/survey.html [hunk 1]
```diff
@@ -61,6 +63,9 @@ <h2 class="saturn u-pt-m">Collection exercises for {{survey.shortName}}</h2>
               <td class="table--cell tbl-ce-respondent-info" name="tbl-ce-shown-as">
                 {{exercise.userDescription}}
               </td>
+              <td class="table--cell tbl-ce-status" name="tbl-ce-status">
+                {{exercise.state}}
```
Comment (tej99) [inline response_operations_ui/templates/survey.html]: You will need to map the the states to a more user readable state:
INIT --> Created
PENDING --> Scheduled
VALIDATED, EXECUTED, PUBLISHED --> Ready for Review, Ready for Live, Live

I would double check this with both the rad team and the devs developing the backend state transition.

Comment (emilioward) [reply response_operations_ui/templates/survey.html]: Yeah i've spoken with Rachel, looks like we need to have a session around the mapping of the backend to frontend states, as these might be incorrect now

**Comment by emilioward** [comment]
[US011 Display State of a Collection Exercise [M]](https://trello.com/c/kJO9QkLd/29-us011-display-state-of-a-collection-exercise-m)

File: response_operations_ui/views/surveys.py [hunk 2]
```diff
@@ -34,7 +34,26 @@ def view_survey(short_name):
             "title": f"{survey_details['survey']['surveyRef']} {survey_details['survey']['shortName']}",
         }
     ]
+
+    # Mapping backend states to frontend sates for the user
+    for collection_exercise_state in survey_details["collection_exercises"]:
+        if collection_exercise_state["state"] == "INIT":
```
Comment (dcdarrell9) [inline response_operations_ui/views/surveys.py]: Are any of these likely to change?

Comment (jcox-dev) [reply response_operations_ui/views/surveys.py]: @emilioward @dcdarrell9 I think @madeye-matt is changing a lot of the state definitions for collection exercises. Are these up to date with the new designs (on Trello)?

Comment (emilioward) [reply response_operations_ui/views/surveys.py]: I mapped these against what I saw in the trello design, but i'm not sure if they were going to change again further, this is something I need to clarify

File: response_operations_ui/views/surveys.py [hunk 3]
```diff
@@ -34,7 +34,26 @@ def view_survey(short_name):
...
+            collection_exercise_state["state"] = "Scheduled"
+
+        if collection_exercise_state["state"] == "READY_FOR_REVIEW":
+            collection_exercise_state["state"] = "Ready for Review"
+
+        if collection_exercise_state["state"] == "EXECUTION_STARTED":
+            collection_exercise_state["state"] = "Processing"
+
+        if collection_exercise_state["state"] == "READY_FOR_LIVE":
+            collection_exercise_state["state"] = "Live"
```
Comment (davidmort) [inline response_operations_ui/views/surveys.py]: I think this should be "Ready for live"

Comment (davidmort) [inline response_operations_ui/views/surveys.py]: we need a mapping for the live status

File: response_operations_ui/views/surveys.py [hunk 4]
```diff
@@ -34,7 +34,26 @@ def view_survey(short_name):
             "title": f"{survey_details['survey']['surveyRef']} {survey_details['survey']['shortName']}",
         }
     ]
+
+    # Mapping backend states to frontend sates for the user
+    for collection_exercise_state in survey_details["collection_exercises"]:
```
Comment (davidmort) [inline response_operations_ui/views/surveys.py]: we need to make this status mapping common as the same mapping is used in the collection exercise details page

**Comment by jcox-dev** [comment]
[US040: Create and View 'Created' State of a CE [M]](https://trello.com/c/TI7VB3Jr/34-us040-create-and-view-created-state-of-a-ce-m)

**Pull request merged**